testcafe/tests/drive_sharing.js

```diff
@@ -0,0 +1,127 @@
+import { Selector, Role } from 'testcafe'
+import { driveUser } from './helpers/roles'
+import {
+  TESTCAFE_DRIVE_URL,
+  isExistingAndVisibile,
+  getCurrentDateTime,
+  FOLDER_DATE_TIME
+} from './helpers/utils'
+import Data from './helpers/data'
+const data = new Data()
+
+import DrivePage from './pages/drive-model'
+const drivePage = new DrivePage()
+
+import PublicDrivePage from './pages/drive-model-public'
+const publicDrivePage = new PublicDrivePage()
+
+const file = '[cozy]QA_table_ APPS.pdf'
+
+//************************
+//Tests when authentified
+//************************
+fixture`Folder link Sharing Scenario`.page`${TESTCAFE_DRIVE_URL}/`.beforeEach(
+  async t => {
+    await t.useRole(driveUser)
+    await drivePage.waitForLoading()
+  }
+)
+
+test('Drive : Create a $test_date_time folder in Drive', async t => {
+  await drivePage.addNewFolder(FOLDER_DATE_TIME)
+  //We need to pass FOLDER_DATE_TIME through multiple fixture, so we cannot use ctx here.
+})
+
+test('Drive : from Drive, go in a folder, upload a file, and share it', async t => {
+  await drivePage.goToFolder(FOLDER_DATE_TIME)
+  await drivePage.openActionMenu()
+  await t.pressKey('esc') //close action Menu
+
+  await drivePage.uploadFiles([`../data/${file}`])
+  await drivePage.shareFolderPublicLink()
+
+  data.sharingLink = await drivePage.copyBtnShareByLink.getAttribute(
+    'data-test-data'
+  )
+  console.log(`SHARING_LINK : ` + data.sharingLink)
+})
+
+//************************
+// Public (no authentification)
+//************************
+fixture`Drive : Access a folder public link`
+  .page`${TESTCAFE_DRIVE_URL}/`.beforeEach(async t => {
+  await t.useRole(Role.anonymous())
+})
+test('Drive : Access a folder public link', async t => {
+  await t.navigateTo(data.sharingLink)
+  await publicDrivePage.waitForLoading()
+
+  await publicDrivePage.checkActionMenuPublicDesktop()
+  await t
+    .setNativeDialogHandler(() => true)
+    .click(publicDrivePage.btnPublicDownload)
+    .click(publicDrivePage.btnPublicCreateCozy)
+  await publicDrivePage.checkCreateCozy()
+
+  //there is some errors if I check for file just after cliking the download button, but not if we do it at the end.
+  await publicDrivePage.checkDownload('files.zip')
+
+  //Mobiles Checks
+  await t.resizeWindowToFitDevice('iPhone 6', {
+    portraitOrientation: true
+  })
+  await publicDrivePage.checkActionMenuPublicMobile()
+  await t
+    .setNativeDialogHandler(() => true)
+    .click(publicDrivePage.btnPublicMobileDownload)
+    .click(publicDrivePage.btnPublicMoreMenu) //need to re-open the more menu
+    .click(publicDrivePage.btnPublicMobileCreateCozy)
+  await publicDrivePage.checkCreateCozy()
+
+  await publicDrivePage.checkDownload('files (1).zip')
+
+  await t.resizeWindow(1280, 1024) //Back to desktop
```

Comment:
Maybe we should do this in a `before` hook?

Reply:
`after` hook seems better as it runs at the end of the test!